### PR TITLE
Add new python script to periodically capture images with the camera

### PR DESCRIPTION
**Pull request by Soso0400**

### CODE REVIEW

File: ImageCapture/webcam_image_capture.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import cv2
+import os
+import time
+from datetime import datetime
+
+# Paramètres
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:6]: Please write comments in english as the project is in english.

File: ImageCapture/webcam_image_capture.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# Création du dossier pour stocker les images, s'il n'existe pas
+if not os.path.exists(save_directory):
+    os.makedirs(save_directory)
+
+# Initialisation de la caméra (0 pour la caméra par défaut, essayez 1 si vous utilisez une autre webcam)
+cap = cv2.VideoCapture(0)
+
+if not cap.isOpened():
+    print("Erreur: Impossible d'accéder à la webcam.")
+    exit()
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:19]: You can add a parameter to the exit function. It is the status code. Usually 0 = OK, 1 = Error

File: ImageCapture/webcam_image_capture.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+interval_seconds = 5  # Intervalle de temps entre chaque capture en secondes
+save_directory = "captured_images"  # Dossier où les images seront enregistrées
+
+# Création du dossier pour stocker les images, s'il n'existe pas
+if not os.path.exists(save_directory):
+    os.makedirs(save_directory)
+
+# Initialisation de la caméra (0 pour la caméra par défaut, essayez 1 si vous utilisez une autre webcam)
+cap = cv2.VideoCapture(0)
+
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:16]: When I try it on my computer, the 5 first pictures are the same. 
You can reduce the size of the VideoCapture buffer by using this line `cap.set(cv2.CAP_PROP_BUFFERSIZE, 1)`.
It means that the two first images will be the same. To fix that I duplicated line 25 (` ret, frame = cap.read()
`) and now all pictures are taken on time.
Could you try that on your side please?

File: ImageCapture/webcam_image_capture.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+if not cap.isOpened():
+    print("Erreur: Impossible d'accéder à la webcam.")
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:18]: Write errors in English please ;)

File: ImageCapture/webcam_image_capture.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    exit()
+
+try:
+    print("Press Ctrl+C to stop capturing images.")
+    while True:
+        # Capture de l'image
+        ret, frame = cap.read()
+        if not ret:
+            print("Erreur: Impossible de lire une frame.")
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:27]: Write errors in English please ;)

File: ImageCapture/webcam_image_capture.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+cap = cv2.VideoCapture(0)
+
+if not cap.isOpened():
+    print("Erreur: Impossible d'accéder à la webcam.")
+    exit()
+
+try:
+    print("Press Ctrl+C to stop capturing images.")
+    while True:
+        # Capture de l'image
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:24]: Write comments in English please ;)

File: ImageCapture/webcam_image_capture.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        ret, frame = cap.read()
+        if not ret:
+            print("Erreur: Impossible de lire une frame.")
+            break
+
+        # Génération d'un nom unique pour l'image avec horodatage
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:30]: Write comments in English please ;)

File: ImageCapture/webcam_image_capture.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+        image_path = os.path.join(save_directory, f"image_{timestamp}.jpg")
+
+        # Sauvegarde de l'image
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:34]: Write comments in English please ;)

File: ImageCapture/webcam_image_capture.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+        cv2.imwrite(image_path, frame)
+        print(f"Image capturée et enregistrée : {image_path}")
+
+        # Pause pour l'intervalle spécifié
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:38]: Write comments in English please ;)

File: ImageCapture/webcam_image_capture.py [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+            print("Erreur: Impossible de lire une frame.")
+            break
+
+        # Génération d'un nom unique pour l'image avec horodatage
+        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+        image_path = os.path.join(save_directory, f"image_{timestamp}.jpg")
+
+        # Sauvegarde de l'image
+        cv2.imwrite(image_path, frame)
+        print(f"Image capturée et enregistrée : {image_path}")
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:36]: Console messages should be written in English

File: ImageCapture/webcam_image_capture.py [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+        # Pause pour l'intervalle spécifié
+        time.sleep(interval_seconds)
+
+except KeyboardInterrupt:
+    print("Capture d'images arrêtée par l'utilisateur.")
+
+# Libération de la caméra et fermeture des fenêtres
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:44]: Write comments in English please ;)

File: ImageCapture/webcam_image_capture.py [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    while True:
+        # Capture de l'image
+        ret, frame = cap.read()
+        if not ret:
+            print("Erreur: Impossible de lire une frame.")
+            break
+
+        # Génération d'un nom unique pour l'image avec horodatage
+        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+        image_path = os.path.join(save_directory, f"image_{timestamp}.jpg")
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:32]: If the period is less than 1 second, what happens? The first image is overridden?

File: ImageCapture/webcam_image_capture.py [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    os.makedirs(save_directory)
+
+# Initialisation de la caméra (0 pour la caméra par défaut, essayez 1 si vous utilisez une autre webcam)
+cap = cv2.VideoCapture(0)
+
+if not cap.isOpened():
+    print("Erreur: Impossible d'accéder à la webcam.")
+    exit()
+
+try:
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:21]: Nice error handling

File: ImageCapture/webcam_image_capture.py [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    print("Press Ctrl+C to stop capturing images.")
+    while True:
```
Comment (BloodFutur) [inline ImageCapture/webcam_image_capture.py:23]: I wonder if there is a better way to periodically call this function. `while True` loops and `time.sleep` can make unexpected behaviors in some cases.

**Review by BloodFutur — CHANGES_REQUESTED** [review]
It is great but in my case there is a buffering problem. Check my comment to fix the problem. Try it on your side to check whether it creates new bugs.

File: ImageCapture/webcam_image_capture.py (hunk 13 above)
Comment (Soso0400) [reply ImageCapture/webcam_image_capture.py:25]: thank you :)

**Review by BloodFutur — CHANGES_REQUESTED** [review]
Very good. Fix overridden pictures behavior when the period is < 1s and it will be perfect.

**Review by BloodFutur — APPROVED** [review]

**Pull request merged**